File: JS/implement/stack.js
"use strict";

const fs = require("fs");

const input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().split("\n")
    : `7
pop
top
push 123
top
pop
top
pop`.split("\n");

const N = parseInt(input[0]);

const stack = [];
const result = [];

for (let i = 1; i < N + 1; i++) {
  const command = input[i].split(" ");

  switch (command[0]) {
    case "push": {
      stack.push(parseInt(command[1]));
      break;
    }
    case "pop": {
      result.push(stack.pop() || -1);
      break;
    }
    case "size": {
      result.push(stack.length);
      break;
    }
    case "empty": {
      result.push(stack.length ? 0 : 1);
      break;
    }
    case "top": {
      result.push(stack[stack.length - 1] || -1);
      break;
    }
    default:
      continue;
  }
}

console.log(result.join("\n"));

File: src/implement/Stack.java
package implement;

import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class Stack {
    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        List<Integer> stack = new ArrayList<>();
        StringBuilder sb = new StringBuilder();

        for(int i = 0; i < N; i++){
            String[] s = br.readLine().split(" ");

            switch (s[0]){
                case "push":{
                    stack.add(Integer.parseInt(s[1]));
                    break;
                }
                case "pop":{
                    if(stack.size() == 0){
                        sb.append(-1 + "\n");
                    }else{
                        sb.append(stack.remove(stack.size() - 1) + "\n");
                    }

                    break;
                }
                case "size": {
                    sb.append(stack.size() + "\n");
                    break;
                }
                case "empty": {
                    if(stack.isEmpty()) sb.append(1 + "\n");
                    else sb.append(0 + "\n");

                    break;
                }
                case "top": {
                    if(stack.isEmpty()) sb.append(-1 + "\n");
                    else sb.append(stack.get(stack.size() - 1) + "\n");

                    break;
                }
                default: continue;
            }
        }

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }
}
